### PR TITLE
Some fixes to make it run

### PR DESCRIPTION
**Pull request by fernflower**
- Address an inpust typo
- Address env_artifacts / artifacts typo in filename
- Use tft api url from inputs
- Add shell:bash to get rid of 'Required property is missing: shell'
- Add 2 more inputs - github_token and api_url - and refactor action
to mitigate 'Unrecognized named-value: 'secrets'
- Add missed $ to plans regex
- Use test_fmf_ref and tests_tmt_tef properly
- Pass artifacts, env_secrets and variables without ""
- Fix schema validation error
- Construct TF_URL from artifacts_url and id
- Use if from github action
- Set defaults for api_url and artifacts_url

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -6,12 +6,18 @@ branding:
   color: red
 
 inputs:
+  api_url:
+    description: 'A testing farm server url'
+    required: true
```
Comment (phracek) [inline action.yml:11]: default would be grade for api_url.

**Review by phracek — COMMENTED** [review]
As soon as it is rebased, we can marge it.

File: action.yml (hunk 1 above)
Comment (fernflower) [reply action.yml:11]: You mean put there default like `https://api.dev.testing-farm.io/v0.1`?

Comment (phracek) [reply action.yml:11]: I guess, this one URL is correct: `https://api.testing-farm.io`

Comment (fernflower) [reply action.yml:11]: Hmmm this one doesn't resolve.

**Comment by fernflower** [comment]
It is failing in wait-till-result stage with 
```

Run cat << EOF > final.json
##[debug]/usr/bin/bash --noprofile --norc -e -o pipefail /home/runner/work/_temp/da09de83-f62e-4830-8c55-fc5aef8647c3.sh
{
  "sha": "943164d5179fad46e5f888a3f4f15ce343be95b1",
  "state": "",
  "context": "Testing Farm - Fedora",
  "description": "Build finished",
  "target_url": "http://artifacts.osci.redhat.com/testing-farm/0094a086-2e70-4759-8081-413beffca31d/0094a086-2e70-4759-8081-413beffca31d"
}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   276    0     0  100   276      0   1445 --:--:-- --:--:-- --:--:--  1445
curl: (52) Empty reply from server
Error: Process completed with exit code 52.
```

Latest run - https://github.com/fernflower/leapp/runs/5156422912?check_suite_focus=true

**Comment by fernflower** [comment]
Sample run (blocking till test results received) - https://github.com/fernflower/leapp/runs/5159543475?check_suite_focus=true

**Comment by fernflower** [comment]
Okay, this should be it - all fixes to make this action usable by leapp team. Unfortunately we can't use status update on the spot because our workflow code has to be refactored (we schedule 2 requests to testing farm in one workflow atm, yeah).

**Review by phracek — APPROVED** [review]
Thanks for a lot of fixes and catching typos. LGTM

**Pull request merged**